packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -116,89 +139,109 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
         color.withOpacity(color.opacity * fadeoutOpacityAnimation.value);
   }
 
-  double _getThumbX(Size size) {
-    assert(textDirection != null);
-    switch (textDirection) {
-      case TextDirection.rtl:
-        return crossAxisMargin;
-      case TextDirection.ltr:
-        return size.width - thickness - crossAxisMargin;
-    }
-    return null;
-  }
+  void _paintThumbCrossAxis(Canvas canvas, Size size, double thumbOffset, double thumbExtent, AxisDirection direction) {
+    double x, y;
+    Size thumbSize;
 
-  void _paintVerticalThumb(Canvas canvas, Size size, double thumbOffset, double thumbExtent) {
-    final Offset thumbOrigin = Offset(_getThumbX(size), thumbOffset);
-    final Size thumbSize = Size(thickness, thumbExtent);
-    final Rect thumbRect = thumbOrigin & thumbSize;
-    if (radius == null)
-      canvas.drawRect(thumbRect, _paint);
-    else
-      canvas.drawRRect(RRect.fromRectAndRadius(thumbRect, radius), _paint);
-  }
+    switch (direction) {
+      case AxisDirection.down:
+        thumbSize = Size(thickness, thumbExtent);
+        x = textDirection == TextDirection.rtl
+          ? crossAxisMargin + padding.left
+          : size.width - thickness - crossAxisMargin - padding.right;
+        y = thumbOffset;
+        break;
+      case AxisDirection.up:
+        thumbSize = Size(thickness, thumbExtent);
+        x = textDirection == TextDirection.rtl
+        ? crossAxisMargin + padding.left
+        : size.width - thickness - crossAxisMargin - padding.right;
+        y = thumbOffset;
+        break;
+      case AxisDirection.left:
+        thumbSize = Size(thumbExtent, thickness);
+        x = thumbOffset;
+        y = size.height - thickness - crossAxisMargin - padding.bottom;
+        break;
+      case AxisDirection.right:
+        thumbSize = Size(thumbExtent, thickness);
+        x = thumbOffset;
+        y = size.height - thickness - crossAxisMargin - padding.bottom;
+        break;
+    }
 
-  void _paintHorizontalThumb(Canvas canvas, Size size, double thumbOffset, double thumbExtent) {
-    final Offset thumbOrigin = Offset(thumbOffset, size.height - thickness);
-    final Size thumbSize = Size(thumbExtent, thickness);
-    final Rect thumbRect = thumbOrigin & thumbSize;
+    final Rect thumbRect = Offset(x, y) & thumbSize;
     if (radius == null)
       canvas.drawRect(thumbRect, _paint);
     else
       canvas.drawRRect(RRect.fromRectAndRadius(thumbRect, radius), _paint);
   }
 
   void _paintThumb(
+    double beforeInset,
     double before,
     double inside,
+    double afterInset,
     double after,
     double viewport,
     Canvas canvas,
     Size size,
-    void painter(Canvas canvas, Size size, double thumbOffset, double thumbExtent),
+    AxisDirection direction,
   ) {
+    final double totalInset = beforeInset + afterInset;
```

Comment:
I think we already have too many terminologies that really mean the same thing between padding, margin and inset. Try to keep it tighter.